src/audit-log/audit-log.ts

```diff
@@ -1,10 +1,17 @@
+import { CreateEventOptions } from './interfaces/create-event-options.interface';
 import { Event } from '../common/interfaces';
 import WorkOS from '../workos';
 
 export class AuditLog {
   constructor(private readonly workos: WorkOS) {}
 
-  async createEvent(event: Event) {
-    await this.workos.post('/events', event);
+  async createEvent(event: Event, { idempotencyKey }: CreateEventOptions = {}) {
+    const headers: any = {};
```

Comment:
worth having a header type? could even just specify known headers since we're building them here

```
const headers: { Idempotency-Key?: string } = {};
```

Reply:
I like this. It's more strict. Will implement.

1.